src/Controller/CRUDController.php

```diff
@@ -1359,15 +1361,51 @@ protected function handleXmlHttpRequestErrorResponse(Request $request, FormInter
             return $this->renderJson([], Response::HTTP_NOT_ACCEPTABLE);
         }
 
-        $errors = [];
-        foreach ($form->getErrors(true) as $error) {
-            $errors[] = $error->getMessage();
+        return $this->json(
```

Comment:
This change from renderJson has 2 implications:

1. IMO we should change all renderJson to simply json and deprecate renderJson (it comes from AbstractController) (that would simplify CRUDController even further. DO you agree? Should I do in separate PR?
2. Here is necessary because we need the behavior of JSON. If you have serializer installed it will use it to serialize classes, in this case the ConstraintVIolationList. Should we add the hard dep here? I don't think so, but maybe we should add it to require-dev? Not sure really, also we should document this change somehow.

@VincentLanglet

Reply:
1) yes might be better
2) i dont think we need the dep so far ; but yes doc/ugrade note is needed

Reply:
added upgrade note, about the change from renderJson to json, it should be handled on another PR.